mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -116,7 +119,13 @@ public void configureJobConf(TableDesc tableDesc, JobConf jobConf) {
       Preconditions.checkArgument(!tableDesc.getTableName().contains(TABLE_NAME_SEPARATOR),
           "Can not handle table " + tableDesc.getTableName() + ". Its name contains '" + TABLE_NAME_SEPARATOR + "'");
       String tables = jobConf.get(InputFormatConfig.OUTPUT_TABLES);
-      tables = tables == null ? tableDesc.getTableName() : tables + TABLE_NAME_SEPARATOR + tableDesc.getTableName();
+      String catalogName = tableDesc.getProperties().getProperty(InputFormatConfig.CATALOG_NAME);
+      if (catalogName != null) {
+        tables = tables == null ? catalogName + CATALOG_NAME_SEPARATOR + tableDesc.getTableName() :
```

Comment:
can we extract the common `catalogName + CATALOG_NAME_SEPARATOR + tableDesc.getTableName()` part into a variable? might make it a bit less verbose

Reply:
done